src/olympia/editors/helpers.py

```diff
@@ -578,7 +579,9 @@ def get_actions(self, request, addon):
                                      'label': label}
             # An unlisted sideload add-on, which requests a full review, cannot
             # be granted a preliminary review.
-            if addon.is_listed or self.review_type == 'preliminary':
+            prelim_allowed = not waffle.flag_is_active(
+                request, 'no-prelim-review') and addon.is_listed
+            if prelim_allowed or self.review_type == 'preliminary':
```

Comment:
Yeah, this doesn't stop existing preliminary add-ons with a new version pending from continuing to be preliminary reviewed, it just stops any more add-ons from becoming preliminary reviewed.  #3087 is dealing with the existing ones.